### ISSUE TITLE
Possible to simplify ParallelWorkItemDispatcher

### ISSUE THREAD

**Issue by oznetmaster** [nunit/nunit#1220]
Since every test now runs on it own thread, there seems to be little reason to continue to maintain separate queues for STA and MTA apartment threaded tests.

Having only a single set of queues would be easier to maintain.  The housekeeping code can run on a default apartment thread.

**Comment by CharliePoole** [comment]
As I believe we already discussed, NUnit has historically run tests on their own thread. We started on a path away from that when we began development of 3.0. You found that cancellation was not working and backed out the change in several (not actually all) instances. I see this as a temporary regression, done as a matter of expediency. As I recorded on the other issue, I expect to get us back on track for the original design. Eliminating the queues in question would serve no purpose and would get us farther away from the direction we are trying to go.

As I also stated in the other issue, I will attempt to recover the design docs that didn't get transfered to the new dev wiki when we created it.
